frontend/packages/kubevirt-plugin/integration-tests/tests/vmtemplate.wizard.scenario.ts

```diff
@@ -174,5 +176,14 @@ describe('Create VM from Template using wizard', () => {
       await vmTemplate.listViewAction(VMT_ACTION.Create);
       await wizard.processWizard(vmCfg);
     });
+
+    it('ID(CNV-4290) Creates VM using VM Template create virtual machine link', async () => {
+      const vmCfg = getVMCfg('vm-from-vmt-createlink');
+
+      vm = new VirtualMachine(vmCfg);
+      await vmTemplate.navigateToListView();
+      await clickCreateVMLink();
```

Comment:
I think it would be better to add this clickCreateVMLink method to the VirtualMachineTemplate class, where we would handle selecting the appropriate row and within the row we can then select the `Create Virtual Machine` link. This way it only works when there is only one VM Template present in the entire list of Virtual Machine Templates.

My suggestion is to add:
vm.list.view.ts:
```
export const rowForUID = (UID: string) => $(`[data-id="${UID}"]`);
```

utils.ts:
```
export function getResourceUID(kind: string, name: string, namespace: string): string {
  return execSync(
    `kubectl get --ignore-not-found ${kind} ${name} -n ${namespace} -o jsonpath='{.metadata.uid}'`,
  ).toString();
}
```

template.view.ts:
```
export const templateCreateVMLink = (templateUID: string) => rowForUID(templateUID).element(by.linkText('Create Virtual Machine'));
```

virtualMachineTemplate.ts:
```
async createVirtualMachineRowLink() {
  const uid = getResourceUID(this.model.kind, this.name, this.namespace);
  await click(templateCreateVMLink(uid));
}
```

Let me know if you hit some issue, I haven't tested the code.

Reply:
Updated like this.